src/nu/validator/htmlparser/impl/TreeBuilder.java

```diff
@@ -1458,6 +1460,17 @@ public final void startTag(ElementName elementName,
         flushCharacters();
 
         // [NOCPP[
+        if (contextNamespace == "http://www.w3.org/1999/xhtml" &&
```

Comment:
`contextNamespace` is the namespace of the fragment parsing context node, so it's the wrong thing here.

Reply:
> `contextNamespace` is the namespace of the fragment parsing context node, so it's the wrong thing here.

Yeah, I subsequently also realized that after testing and looking through the code more…

Anyway, that’s gone now — since we now just set the voidElement flage per-element (avoiding the foreign elements)